sympy/vector/vector.py

```diff
@@ -40,6 +40,34 @@ def components(self):
         # subclass of Vector the instance belongs to.
         return self._components
 
+    @property
+    def projections(self):
+        """
+        Returns the components of this vector but the output includes
+        also zero values components.
+
+        Examples
+        ========
+
+        >>> from sympy.vector import CoordSys3D, Vector
+        >>> C = CoordSys3D('C')
+        >>> v1 = 3*C.i + 4*C.j + 5*C.k
+        >>> v1.projections
+        {C.i: 3, C.j: 4, C.k: 5}
+        >>> v2 = C.x*C.y*C.z*C.i
+        >>> v2.projections
+        {C.i: C.x*C.y*C.z, C.j: 0, C.k: 0}
+        >>> v3 = Vector.zero
+        >>> v3.projections
+        0
+        """
+
+        from sympy.vector.operators import _get_coord_sys_from_expr
+        if self is Vector.zero:
+            return S(0)
+        base_vec = next(iter(_get_coord_sys_from_expr(self))).base_vectors()
+        return dict(zip(base_vec, [self.components[i] if i in self.components else S(0) for i in base_vec]))
```

Comment:
Maybe instead of `self.components[i]` you should use `dot`. I made some edits that make `.components` incomplete in case of unevaluated expressions.